Source/Xamarin/Prism.Forms/Navigation/PageNavigationService.cs

```diff
@@ -475,18 +475,21 @@ void ConfigureTabbedPage(TabbedPage tabbedPage, string segment)
 
                 foreach (var child in tabbedPage.Children)
                 {
-                    if (child.GetType() != selectedTabType)
-                        continue;
+                    SetAutowireViewModelOnPage(child);
 
                     if (child is NavigationPage)
                     {
+                        child.Behaviors.Add(new Behaviors.NavigationPageActiveAwareBehavior());
+
                         var childTabType = PageNavigationRegistry.GetPageType(UriParsingHelper.GetSegmentName(selectedTabChildSegment));
                         if (((NavigationPage)child).CurrentPage.GetType() != childTabType)
                             continue;
                     }
 
+                    if (child.GetType() != selectedTabType)
+                        continue;
```

Comment:
You could get rid of the continue and just make this:

```cs
if(child.GetType() == selectedTabType)
    tabbedPage.CurrentPage = child;
```

Reply:
Actually, I need to rethink that. I changed the order of some of the snippets, and I'm not sure they'll work as expected.  I need to write more tests to confirm :)